tools/codegen/codegen.py

```diff
@@ -104,6 +105,20 @@ def _merge(a, b, path=None):
         raise TypeError(f"Could not merge type {type(a)!r} with type {type(b)!r}.")
     return a
 
+def log_filter(text):
+    for frameinfo in inspect.stack():
+        template = frameinfo.frame.f_globals.get("__jinja_template__")
+        if template is not None:
+            break
+    lineno = 0
```

Comment:
nit: return early and avoid an if/else by checking if template is nil first

Reply:
I think I misunderstand you.  What's the specific change you're suggesting here?